docs/playground/README.md

```diff
@@ -0,0 +1,90 @@
+
+The wallet pane is only active after the current contract has been compiled successfully. In fact, only in this case is the correct set of wallet endpoints known. Execution of the event sequence may fail when one or more arguments to one or more events is not a legitimate value of the corresponding type of contract endpoint function. In this case, each event with malformed argument(s) ought to be tinted red, while all malformed argument(s) are highlighted.
+
+### Mockchain pane
+
+The mockchain pane has three major parts.
+
+1. A visual representation of the mockchain in the form of the UTxO graph. The UTxO graph consists of nodes that represent individual (valid) transactions and edges that represent (valid) input-output connections.
+2. The final assignment of funds to wallets. This may be a subsets of the overall funds, because some funds may be locked by scripts. Optionally, the wallets should be colour coded and edges paying funds to those wallets should use the same colour. Moreover, we may display the wallet funds as coloured bars whose relative length represents their relative (final) value.
+3. A possibly empty list of *invalid* transactions, each of which indicates the reason for why it is invalid.
+
+Optionally, provide introspection facilities for valid and invalid transactions and graph edges. In particular, to query information that helps with debugging and also includes rejected, invalid transactions and failed script execution.
+
+
+## Protocol between the frontend and backend
+
+We are using a stateless architecture. Hence, the entire frontend state (that affects contract execution) needs to be transmitted to the backend on each request. Central is the contract source code and the current sequence of *wallet events*. A wallet event comprises the name of a *wallet endpoint function* and a list of arguments matching the type of the wallet endpoint function. Wallet endpoint functions are those designated functions within a contract that can be invoked by the user of a wallet that has loaded the contract.
+
+### Frontend state required for contract execution
+
+1. The contents of all Plutus source modules. (Initially, this is one module; optionally, we will extend it to a list of modules subsequently.)
+2. List of initial amount of ADA in each wallet. (The length of this list determines the number of wallets.)
+3. List of wallet events.
+
+### Wallet events
+
+The wallet events are invocations of wallet endpoint functions (aka *contract endpoints*), where the wallet user provides arguments to the endpoint function. The user does so within the frontend UI, which requires a limit on the number and form of those arguments. We support the following types as arguments types for wallet endpoint function s:
+
+* `String`,
+* `PubKey`,
+* integral types, and
+* products of the above.
+
+Products can be in the form of record datatypes with a single constructor. Wallet endpoint functions can, at most, have TBD arguments.
```

Comment:
I think in the end we decided it was easy enough to support anything `Generic`?

Reply:
`Generic` is fine for the code, but I am still not sure how to handle this in the frontend. How does the user enter, say, a binary tree value?

Reply:
Can't we let users provide (or derive) Show and Read instances for this purpose? We could do much nicer demos if they could involve arbitrary Haskell data structures, with user-defined Show and Read.